emitSchemaTypes.js

```diff
@@ -61,30 +61,36 @@ const writeTypeFile = (file, commonImport, snippets) => {
   }
 };
 
-const emitSchemaTypes = async file => {
-  try {
-    if (!file.match(/Schema/) || file.match(/spec/)) {
-      return;
-    }
+const writeSchema = async (schemas, file) => {
+  const snippets = await Promise.all(
+    Object.entries(schemas).map(([name, schema]) => {
+      if (!name.match(/Schema$/)) {
+        return '';
+      }
+      return compile(schema, schema.title || firstUp(name), opts);
+    })
+  );
+
+  const contents = fs.readFileSync(file).toString();
 
-    const schemas = require(`./${file}`);
+  writeTypeFile(file, typeImports(contents.match(typeImportFindRegex)), snippets);
+};
```

Comment:
Can you quickly explain what is the purpose of this refactor / change? Thanks.

Reply:
this refactor allows for files without the name convention "*Schema" to be also used to generate types as long as they are inside shared/types, so now the convention is SomethingSchema or whatever filename inside shared/types.
specifically this was made to generate the types for https://github.com/huridocs/uwazi/blob/d2fc5a99c6c6a8525f3f8e76c559d2f445778b0c/app/shared/types/searchParams.ts